yesod-test/Yesod/Test.hs

```diff
@@ -24,8 +24,86 @@ This is very useful for testing web applications built in yesod, for example,
 where your forms may have field names generated by the framework or a randomly
 generated CSRF token input.
 
-Your database is also directly available so you can use 'runDB' to set up
+=== Example project
+
+The best way to see an example project using yesod-test is to create a scaffolded Yesod project:
+
+@stack new projectname yesod-sqlite@
+
+(See https://github.com/commercialhaskell/stack-templates/wiki#yesod for the full list of Yesod templates)
+
+The scaffolded project makes your database directly available in tests, so you can use 'runDB' to set up
 backend pre-conditions, or to assert that your session is having the desired effect.
+It also handles wiping your database between each test.
+
+=== Example code
+
+The code below should give you a high-level idea of yesod-test's capabilities.
+Note that it uses helper functions like @withApp@ and @runDB@ from the scaffolded project; these aren't provided by yesod-test.
+
+@
+spec :: Spec
+spec = withApp $ do
+  describe \"Homepage\" $ do
```

Comment:
Unfortunately escaping Haddock is a pain.

Here's a screenshot showing there aren't any Haddock escaping mistakes (I hope!):

![image](https://user-images.githubusercontent.com/1274145/83362337-4e4d9680-a345-11ea-995b-b24efb24cccb.png)